qemu/tests/rh_kernel_update.py

```diff
@@ -8,175 +7,100 @@
-        rh_kernel_hint = r"[\d+][^\s]+"
-        kernel_re = params.get("kernel_re")
-        tag = params.get("brew_tag")
-        kbuild_name = params.get("kernel_build_name", "kernel")
-
-        latest_pkg_cmd = "brew latest-pkg %s %s" % (tag, kbuild_name)
-        o = process.system_output(latest_pkg_cmd, timeout=360).decode()
-        build = re.findall(r"kernel[^\s]+", o)[0]
-        logging.debug("Latest package on brew for tag %s is %s" %
-                      (tag, build))
-
-        buildinfo = process.system_output("brew buildinfo %s" % build,
-                                          timeout=360).decode()
-
-        # install kernel-firmware
-        firmware_url = None
-        if "firmware" in buildinfo:
-            logging.info("Found kernel-firmware")
-            fw_pattern = ".*firmware.*"
-            try:
-                fw_brew_link = re.findall(fw_pattern, buildinfo)[0]
-            except IndexError:
-                test.error("Could not get kernel-firmware package"
-                           " brew link matching pattern '%s'" % fw_pattern)
-            firmware_url = get_brew_url(fw_brew_link, download_root)
-
-        knl_pattern = kernel_re % rh_kernel_hint
-        try:
-            knl_brew_link = re.findall(knl_pattern, buildinfo, re.I)[0]
-        except IndexError:
-            test.error("Could not get kernel package brew link"
-                       " matching pattern '%s'" % knl_pattern)
-        kernel_url = get_brew_url(knl_brew_link, download_root)
-
-        debug_re = kernel_re % ("(%s)" % rh_kernel_hint)
-        try:
-            kernel_version = re.findall(debug_re, kernel_url, re.I)[0]
-        except IndexError:
-            test.error("Could not get kernel version matching"
-                       " pattern '%s'" % debug_re)
-        kernel_version += "." + params.get("kernel_suffix", "")
-
-        return kernel_version, kernel_url, firmware_url
-
-    def get_kernel_debuginfo_rpm_link():
-        knl_dbginfo_re = params.get("knl_dbginfo_re")
-        tag = params.get("brew_tag")
-        kbuild_name = params.get("kernel_build_name", "kernel")
-
-        latest_pkg_cmd = "brew latest-pkg %s %s" % (tag, kbuild_name)
-        o = process.system_output(latest_pkg_cmd, timeout=360).decode()
-        build = re.findall(r"kernel[^\s]+", o)[0]
-        logging.debug("Latest package on brew for tag %s is %s" %
-                      (tag, build))
+    def get_brew_latest_pkg(topdir, tag, pkg, arch=None, list_path=False):
+        """
+        Get url of latest packages in brewweb.
 
-        buildinfo = process.system_output("brew buildinfo %s" % build,
-                                          timeout=360).decode()
+        :parm pkg: package name without version info.
+        :parm brew_tag:  requried in cfg file.
+        :parm vm_arch_name: requried in cfg file.
+        :parm latest_pkg_cmd: requried in cfg file.
 
+        :return: urls for pkg in brewweb.
+        :rtype: list
```

Comment:
From my checking, the `return` is the not a urls for pkg, it is the build name, tag, and anthor,  such as:
```
u'kernel-3.10.0-915.el7                     rhel-7.6-candidate    brdeoliv'
```
and the rtype is not a list.